### PR TITLE
add support for zap v15 and update zap version

### PR DESCRIPTION
**Pull request by mschoch**

### CODE REVIEW

**Comment by mschoch** [comment]
Failures expected because of circular deps

File: go.mod [hunk 1]
```diff
@@ -8,10 +8,11 @@ require (
...
 	github.com/blevesearch/snowballstem v0.9.0
-	github.com/blevesearch/zap/v11 v11.0.10
-	github.com/blevesearch/zap/v12 v12.0.10
-	github.com/blevesearch/zap/v13 v13.0.2
-	github.com/blevesearch/zap/v14 v14.0.1
+	github.com/blevesearch/zap/v11 v11.0.11
+	github.com/blevesearch/zap/v12 v12.0.11
+	github.com/blevesearch/zap/v13 v13.0.3
+	github.com/blevesearch/zap/v14 v14.0.2
+        github.com/blevesearch/zap/v15 v15.0.0
```
Comment (abhinavdangeti) [inline go.mod:15]: Indentation here and below looks off.

**Review by abhinavdangeti — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (mschoch) [reply go.mod]: Yeah I copy/pasted and it converted tabs to spaces.  Pushed a commit using tab.

File: index/scorch/segment_plugin.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ import (
...
 	zapv14 "github.com/blevesearch/zap/v14"
+	zapv15 "github.com/blevesearch/zap/v15"
 )
 
 var supportedSegmentPlugins map[string]map[uint32]segment.Plugin
 var defaultSegmentPlugin segment.Plugin
 
 func init() {
 	ResetPlugins()
+	RegisterPlugin(zapv15.Plugin(), false)
```
Comment (tmm1) [inline index/scorch/segment_plugin.go:34]: Am I reading this correctly that zap v11 is still used by default? How does one tell bleve to use v15?

Comment (mschoch) [reply index/scorch/segment_plugin.go:34]: Yes, we made a decision earlier that the defaults are essentially part of the API, so changing the file format would be equivalent to a breaking change.

We are [planning bleve 2.x now](https://github.com/blevesearch/bleve/issues/1495), and scorch will become the default index type, and the latest zap will also become the default at that time.

One can force the segment type and version used with the `forceSegmentType` and `forceSegmentVersion` config settings.  They are described here: https://github.com/blevesearch/bleve/issues/1350

**Pull request merged**